Fix NPC loot

angelof9 opened the pull request:
This PR should fix:
- NPC loot
- Unidentified gear/material loot

angelof9 committed:
Fix NPC loot
Fix Unidentified gear/material loot

ZeromusXYZ reviewed (commented):
I don't think these are supposed to drop double (it does this for all drops)
![image](https://github.com/user-attachments/assets/09c2cc96-dadf-4e13-9f36-952066b27293)

![image](https://github.com/user-attachments/assets/bdb339a2-0c14-4c75-8d2c-0504844c54f6)

ZeromusXYZ reviewed (changes requested):
Needs some changes to at least fix the double drop issue.

angelof9 committed:
Fix loots again :)
- Fix double loots
- Normalized rates distribution per loot group
- Add cumulative rates to avoid random choice in loot group
- Handle multiple Quests items in groups > 0

angelof9 commented:
New fix should work fine.

Logic is:

```
for each groups in lootpack
    get group dropRate
    if group <=0
        for each loot in group
            do actability
            check for quest item
            roll item
            if success
                add to selected items
    else
        roll for group
        if success
            make tmp lists (quest & regular loot)
            for each loots in group
                check for quest item
                true
                    add to tmpQuestItems
                false
                    add to tmpRegularItems
                    calculate normalizedRate
            roll for item
            for each loot in regularItems (from best chance rates to lower chance rates, ordered by SQL queries)
                add item rate to cumulative rate
                if success
                    add item to selected items
                    break loop
            merge selected quest items to selected items

    for each selected loot
        do random amount
        do grade
        do coin rates
```

angelof9 committed:
AlwaysDrop
- Group=0 => Override item roll for AlwaysDrop
- Group>0 => Set max item rate for each item with AlwaysDrop

angelof9 committed:
ActabilityGroups
Move ActabilityGroups where it should matter.
There is no ActabilityGroups for group=0, so moving it to good location.

ZeromusXYZ reviewed (approved):
Seems to be working correctly now.
Some of the more exotic loot seems to be more rare than it should during tests, but I just blame that on RNGesus.
